src/Desktop/ArcGISRuntime.Desktop.Samples/Samples/Map/AuthorMap/AuthorMap.xaml.cs

```diff
@@ -0,0 +1,497 @@
+        }
+        #endregion
+    }
+
+    #region Helper class to display the OAuth authorization challenge
+    public class OAuthAuthorize : IOAuthAuthorizeHandler
+    {
+        // Window to contain the OAuth UI
+        private Window _window;
+        // Use a TaskCompletionSource to track the completion of the authorization
+        private TaskCompletionSource<IDictionary<string, string>> _tcs;
+        // URL for the authorization callback result (the redirect URI configured for your application)
+        private string _callbackUrl;
+        // URL that handles the OAuth request
+        private string _authorizeUrl;
+
+        // Function to handle authorization requests, takes the URIs for the secured service, the authorization endpoint, and the redirect URI
+        public Task<IDictionary<string, string>> AuthorizeAsync(Uri serviceUri, Uri authorizeUri, Uri callbackUri)
+        {
+            // If the TaskCompletionSource or Window are not null, authorization is in progress
+            if (_tcs != null || _window != null)
+            {
+                // Allow only one authorization process at a time
+                throw new Exception(); 
+            }
+
+            // Store the authorization and redirect URLs
+            _authorizeUrl = authorizeUri.AbsoluteUri;
+            _callbackUrl = callbackUri.AbsoluteUri;
+
+            // Create a task completion source
+            _tcs = new TaskCompletionSource<IDictionary<string, string>>();
+            var tcs = _tcs;
+
+            // Call a function to show the login controls, make sure it runs on the UI thread for this app
+            var dispatcher = Application.Current.Dispatcher;
+            if (dispatcher == null || dispatcher.CheckAccess())
+                AuthorizeOnUIThread(_authorizeUrl);
+            else
+            {
+                dispatcher.BeginInvoke((Action)(() => AuthorizeOnUIThread(_authorizeUrl)));
+            }
+            
+            // Return the task associated with the TaskCompletionSource
+            return tcs.Task;
+        }
+
+        // Challenge for OAuth credentials on the UI thread
+        private void AuthorizeOnUIThread(string authorizeUri)
+        {
+            // Create a WebBrowser control to display the authorize page
+            var webBrowser = new WebBrowser();
+            // Handle the navigation event for the browser to check for a response to the redirect URL
+            webBrowser.Navigating += WebBrowserOnNavigating;
+
+            // Display the web browser in a new window 
+            _window = new Window
+            {
+                Content = webBrowser,
+                Height = 430,
+                Width = 395,
+                WindowStartupLocation = WindowStartupLocation.CenterOwner,
+                Owner = Application.Current != null && Application.Current.MainWindow != null
+                            ? Application.Current.MainWindow
+                            : null
+            };
```

Comment:
Could you add some comments on what is actually happening on this line of code (i.e. the 'Owner' object). Seems like a lot of voodoo magic. I image this will be quite a bear to code in iOS and Android.